### DISCUSSION TITLE
RS485 JDB BMS Not Working

### DISCUSSION THREAD

**Discussion by plloppii** [KG-Development/SmartBMSUtility#29] [General]
Hi there, 

I purchased this app in the mac os app store and I tried using it with my JDB BMS that I purchased here:
https://www.aliexpress.us/item/2255800914539458.html?spm=a2g0o.order_list.order_list_main.200.62281802AKp0gk&gatewayAdapt=glo2usa4itemAdapt

My unit communicates through RS485. I was under the impression that it would just work out of the box, but the app doesnt pick up the BMS. 

Is there any other information that I can provide to possibly add this in a future version?

**Comment by NeariX67** [comment]
Hey, thanks for reaching out!

Currently, we do not support wired communication, however this is going to change withh the upcoming release of version 3, we will feature serial support using the small black box that comes with the BMS.

We are expecting to release a TestFlight Beta this week, so use the link in the Readme to register yourself.
Public release will follow shortly, as far as no major issues are reported.

Thanks :)

**Comment by NeariX67** [comment]
@plloppii The iOS Version has been approved for TestFlight, while the MacOS Version is still waiting for approval.

You can already download TestFlight from the MacOS AppStore and use [this link](https://testflight.apple.com/join/YWdbkZ8s) to enter. You will be notified when the App is available for download, which should happen in the next 24 hours.

**Comment by tcambell** [comment]
A couple issues with the new version:The device name can be changed on the settings page but as soon as you go back to the overview screen it gets named back the the original generic name. All my saved devices got lost with the update. It would be nice if the update would keep my saved devices.The symbols used to show read data and write data are very confusing in the configuration page.The temperature in the configuration page is reading deg F but the unit shown is deg C. This could be dangerous! The temperature units should match what was chosen in the settings.Old version New version Thanks TimOn Aug 4, 2023, at 8:36 AM, Justin Kühner ***@***.***> wrote:﻿
@plloppii The iOS Version has been approved for TestFlight, while the MacOS Version is still waiting for approval.
You can already download TestFlight from the MacOS AppStore and use this link to enter. You will be notified when the App is available for download, which should happen in the next 24 hours.

**Comment by tcambell** [comment]
The device list also used to show the device name and BMS name if you changed it in the configuration page. That was a very nice feature On Aug 4, 2023, at 4:37 PM, Tim Campbell ***@***.***> wrote:﻿A couple issues with the new version:The device name can be changed on the settings page but as soon as you go back to the overview screen it gets named back the the original generic name. All my saved devices got lost with the update. It would be nice if the update would keep my saved devices.The symbols used to show read data and write data are very confusing in the configuration page.The temperature in the configuration page is reading deg F but the unit shown is deg C. This could be dangerous! The temperature units should match what was chosen in the settings.Old version <image0.png>New version <image1.png>Thanks TimOn Aug 4, 2023, at 8:36 AM, Justin Kühner ***@***.***> wrote:﻿
@plloppii The iOS Version has been approved for TestFlight, while the MacOS Version is still waiting for approval.
You can already download TestFlight from the MacOS AppStore and use this link to enter. You will be notified when the App is available for download, which should happen in the next 24 hours.

**Comment by NeariX67** [comment]
@tcambell thank you for your feedback. We have corrected everything that you have noticed. A new testflight version has been published.

**Comment by NeariX67** [comment]
@plloppii Have you been able to test out the serial interface with the MacOS Version?

**Comment by tcambell** [comment]
Hi Justin,Thanks for making the changes. Have you noticed that only the left side of the connect buttons work to select the BMS. On my iPhone this is the case.On Aug 5, 2023, at 2:35 PM, Justin Kühner ***@***.***> wrote:﻿
@tcambell thank you for your feedback. We have corrected everything that you have noticed. A new testflight version has been published.

**Reply by ImVelas** [reply]
Hey @tcambell, thanks for your feedback! Unfortunately we cant reproduce this bug. Maybe you can share some more detailed information so we can try to fix it as soon as possible. Maybe you have a screenrecord or something. 
Regards Fabian :)

**Comment by tcambell** [comment]
I don’t get it. The entire button now works. I guess my phone has issues!TimOn Aug 7, 2023, at 12:59 PM, Fabian ***@***.***> wrote:﻿
Hey @tcambell, thanks for your feedback! Unfortunately we cant reproduce this bug. Maybe you can share some more detailed information so we can try to fix it as soon as possible. Maybe you have a screenrecord or something.
Regards Fabian :)

**Comment by plloppii** [comment]
@NeariX67 Hi there, I was able to test the new version that supports usb, and it is indeed working, however the main feature that I need is to be able to calibrate charging, idle, and discharge current with my bms. Is there anyway that that can be supported?

**Reply by NeariX67** [reply]
Yes, we have planned that feature to be included in v3.0.2, which might take some further time to be released.
